### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by stdavis**
## Description of Changes
This pull requests ...

### Test results and coverage

```
(forklift) λ pytest
============================================================================================ test session starts =============================================================================================
platform win32 -- Python 3.9.18, pytest-8.2.0, pluggy-1.5.0
rootdir: C:\projects\forklift
configfile: pyproject.toml
testpaths: tests, src
plugins: anyio-3.5.0, cov-5.0.0, instafail-0.5.0, mock-3.14.0, ruff-0.3.2
collected 182 items

tests\test_arcgis.py .................                                                                                                                                                               [ 17/182]
tests\test_change_detection.py ..........                                                                                                                                                            [ 27/182]
tests\test_changes.py ......                                                                                                                                                                         [ 33/182]
tests\test_config.py .......                                                                                                                                                                         [ 40/182]
tests\test_core.py ......................................                                                                                                                                            [ 78/182]
tests\test_crate.py ...................                                                                                                                                                              [ 97/182]
tests\test_engine.py ......................s..........                                                                                                                                               [130/182]
tests\test_lift.py .........                                                                                                                                                                         [139/182]
tests\test_messaging.py .....                                                                                                                                                                        [144/182]
tests\test_pallet.py ................................                                                                                                                                                [176/182]
tests\test_seat.py ....                                                                                                                                                                              [180/182]
tests\test_slack.py ..                                                                                                                                                                               [182/182]C:\U
sers\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\coverage\inorout.py:503: CoverageWarning: Module projectname was never imported. (module-not-imported)
  self.warn(f"Module {pkg} was never imported.", slug="module-not-imported")
C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\coverage\control.py:888: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
WARNING: Failed to generate report: No data to report.

C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pytest_cov\plugin.py:339: CovReportWarning: Failed to generate report: No data to report.

  self.cov_controller.finish()


============================================================================================== warnings summary ==============================================================================================
..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_array.py:22
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_array.py:22: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    PANDAS_GE_024 = str(pd.__version__) >= LooseVersion("0.24.0")

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\setuptools\_distutils\version.py:345
..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\setuptools\_distutils\version.py:345
..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\setuptools\_distutils\version.py:345
..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\setuptools\_distutils\version.py:345
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\setuptools\_distutils\version.py:345: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_array.py:23
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_array.py:23: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    PANDAS_GE_025 = str(pd.__version__) >= LooseVersion("0.25.0")

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_array.py:24
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_array.py:24: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    PANDAS_GE_10 = str(pd.__version__) >= LooseVersion("1")

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_dask.py:21
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\arcgis\features\geo\_dask.py:21: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    DASK_2021_06_0 = str(dask.__version__) >= LooseVersion("2021.06.0")

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\jupyter_client\session.py:37
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\jupyter_client\session.py:37: DeprecationWarning: zmq.eventloop.ioloop is deprecated in pyzmq 17. pyzmq now works with default tor
nado and asyncio eventloops.
    from zmq.eventloop.ioloop import IOLoop

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\notebook\utils.py:280
..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\notebook\utils.py:280
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\notebook\utils.py:280: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return LooseVersion(v) >= LooseVersion(check)

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:121
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:121: DeprecationWarning: pkg_resources is deprecated as an API
    warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packag
es
    declare_namespace(pkg)

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('google')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packag
es
    declare_namespace(pkg)

..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870
..\..\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870
  C:\Users\agrc-arcgis\AppData\Local\ESRI\conda\envs\forklift\lib\site-packages\pkg_resources\__init__.py:2870: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('ruamel')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packag
es
    declare_namespace(pkg)

src\forklift\engine.py:12
  C:\projects\forklift\src\forklift\engine.py:12: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import load_source

tests/test_engine.py::TestListPallets::test_pallet_with_import_error
  c:\projects\forklift\tests\test_engine.py:183: DeprecationWarning: Please use assertRegex instead.
    self.assertRegexpMatches(import_error, "expected an indented block")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform win32, python 3.9.18-final-0 -----------

========================================================================== 181 passed, 1 skipped, 18 warnings in 597.23s (0:09:57) ===========================================================================
```

### Speed test results

```
Speed Test Results
Dry Run: 2.37 minutes
Repeat: 2.01 minutes
```

### CODE REVIEW

File: src/forklift/__main__.py [hunk 1]
```diff
@@ -59,19 +59,20 @@
                                                                             use the data that was updated rather than the entire ArcGIS Server instance. Note: this
                                                                             will only work for pallets that are in the warehouse.
     forklift speedtest                                                      Test the speed on a predefined pallet.
-'''
+"""
+
 #: check for Pro license
 try:
-    import arcpy
+    pass
```
Comment (steveoh) [inline src/forklift/__main__.py:66]: **question**: should this have been changed?

**Review by steveoh — COMMENTED** [review]
it's hard to tell what's changed but the first few files look good

File: src/forklift/__main__.py (hunk 1 above)
Comment (stdavis) [reply src/forklift/__main__.py:66]: Great catch! I'll fix this.

**Review by steveoh — APPROVED** [review]

**Comment by steveoh** [comment]
We never did anything with the duplicate removal option right from the title?

**Comment by stdavis** [comment]
> We never did anything with the duplicate removal option right from the title?

Yes, that's correct. I'll fix the title.

**Pull request merged**